### PR TITLE
C22 task-list-api updates

### PR DESCRIPTION
**Pull request by kelsey-steven-ada**
This PR covers several tickets. The main changes are to the scaffold for Flask & SQLAlchemy version updates. 

Asana Tickets:
- [Flask & SQLAlchemy updates](https://app.asana.com/0/1201700438643002/1206697443678562/f)
- [Add info & screenshots for selecting Runtime on Render](https://app.asana.com/0/1205655776549324/1204622316280353/f)
- [Switch Task & Goal models to use `id` attribute name](https://app.asana.com/0/1205655776549324/1203308180549116/f)
- [Incorrect requirement for completed_at in Wave 1](https://app.asana.com/0/1205655776549324/1204536134940759/f)

### CODE REVIEW

File: ada-project-docs/optional-enhancements.md [hunk 1]
```diff
@@ -54,13 +46,13 @@ Here are some ideas to start:
 
 ### Use List Comprehensions
 
-Use list comprehensions in your `routes.py` logic.
+Use list comprehensions in your routes functions where applicable.
```
Comment (kelsey-steven-ada) [inline ada-project-docs/optional-enhancements.md:49]:
```suggestion
Use list comprehensions in your route functions where applicable.
```

File: ada-project-docs/wave_05.md
Comment (yangashley) [inline ada-project-docs/wave_05.md:1]: Another nit unrelated to your changes: Could we update "Wave 01" on line 21 to "Wave 1"?

File: ada-project-docs/setup.md
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply ada-project-docs/setup.md:1]: Yes! I'll do a pass to see if it shows up anywhere else I missed.

File: ada-project-docs/wave_01.md [hunk 2]
```diff
@@ -18,17 +18,17 @@ Our goal for this wave is to be able to create, read, update, and delete differe
...
 - `title`: text to name the task
 - `description`: text to describe the task
-- `completed_at`: a datetime that has the date that a task is completed on. **Can be _nullable_,** and contain a null value. A task with a `null` value for `completed_at` has not been completed. When we create a new task, `completed_at` should be `null` AKA `None` in Python.
+- `completed_at`: a datetime that represents the date that a task is completed on. **Can be _nullable_,** and contain a null value. A task with a `null` value for `completed_at` has not been completed. When we create a new task, `completed_at` should be `null` AKA `None` in Python.
 
 ### Tips
 
-- SQLAlchemy's column type for text is `db.String`. The column type for datetime is `db.DateTime`.
-- SQLAlchemy supports _nullable_ columns with specific syntax.
+- To work with date information, we can import the `datetime` data type with the import line `from datetime import datetime`. `datetime`
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply ada-project-docs/wave_01.md:30]: That's a bonus `datetime` that should be removed.

**Review by yangashley — APPROVED** [review]

**Pull request merged**